graylog2-server/src/test/java/org/graylog/testing/completebackend/apis/Streams.java

```diff
@@ -17,17 +17,25 @@
 package org.graylog.testing.completebackend.apis;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
 import io.restassured.response.ValidatableResponse;
 import io.restassured.specification.RequestSpecification;
 
 import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 
 public final class Streams implements GraylogRestApi {
```

Comment:
It might be a stupid suggestion, as I have not thought about it deeply, but is there a way to provide StreamRouter as a dependency inside this class, instead of adding a new endpoint to the REST resource, which will be visible to anyone, but useful only in tests?

Reply:
It's a good point and I am currently rewriting that part. It will get its own "debug" resource, hidden behind an environment variable. The resource is needed, because in the integration test, the API is the only way how to access the running graylog instance.